lib/srv/regular/sshserver.go

```diff
@@ -419,6 +432,132 @@ func (s *Server) startPeriodicOperations() {
+		Kinds: []types.WatchKind{
+			{
+				Kind: types.KindStaticHostUser,
+			},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer watcher.Close()
+
+	// Fetch any host users that existed prior to creating the watcher.
+	var startKey string
+	for {
+		users, nextKey, err := s.staticHostUsers.ListStaticHostUsers(s.Context(), 0, startKey)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		for _, hostUser := range users {
+			if err := s.handleNewHostUser(hostUser); err != nil {
+				// Log the error so we don't stop the handler.
+				slog.WarnContext(s.Context(), "Error handling static host user.", "error", err, "login", hostUser.GetMetadata().Name)
+			}
+		}
+		if nextKey == "" {
+			break
+		}
+		startKey = nextKey
+	}
+
+	// Listen for new host users on the watcher.
+	for {
+		select {
+		case event := <-watcher.Events():
+			if event.Type != types.OpPut {
+				continue
+			}
+			r, ok := event.Resource.(types.Resource153Unwrapper)
+			if ok {
+				hostUser, ok := r.Unwrap().(*userprovisioningpb.StaticHostUser)
+				if ok {
+					if err := s.handleNewHostUser(hostUser); err != nil {
+						// Log the error so we don't stop the handler.
+						slog.WarnContext(s.Context(), "Error handling static host user.", "error", err, "login", hostUser.GetMetadata().Name)
+						continue
+					}
+				}
+			}
+			return trace.Errorf("unexpected resource type: %T", event.Resource)
+		case <-watcher.Done():
+			if watcher.Error() != nil {
+				return watcher.Error()
+			}
+			return nil
+		case <-s.Context().Done():
+			if !errors.Is(s.Context().Err(), context.Canceled) {
+				return trace.Wrap(s.Context().Err())
+			}
+			return nil
+		}
+	}
+
+}
+
+func (s *Server) handleNewHostUser(hostUser *userprovisioningpb.StaticHostUser) error {
+	var createUser *userprovisioningpb.Matcher
+	for _, matcher := range hostUser.Spec.Matchers {
+		// Check if this host user applies to this node.
+		nodeLabels := make(types.Labels)
+		for k, v := range label.ToMap(matcher.NodeLabels) {
+			nodeLabels[k] = apiutils.Strings(v)
+		}
+		matched, _, err := services.CheckLabelsMatch(
+			types.Allow,
+			types.LabelMatchers{
+				Labels:     nodeLabels,
+				Expression: matcher.NodeLabelsExpression,
+			},
+			nil, // userTraits
+			s.getBasicInfo(),
+			false, // debug
+		)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if !matched {
+			continue
+		}
+
+		// Matching multiple times is an error.
+		if createUser != nil {
+			return trace.BadParameter("Static host user matched this node multiple times. Please update resource to ensure that each node is matched only once.")
+		}
+		createUser = matcher
+	}
+
+	// If we found a match, create the user.
+	if createUser != nil {
+		login := hostUser.GetMetadata().Name
+		slog.DebugContext(s.Context(), "Updating matched static host user", "login", login)
+		ui := services.HostUsersInfo{
+			Groups:       createUser.Groups,
+			Mode:         types.CreateHostUserMode_HOST_USER_MODE_KEEP,
```

Comment:
Not sure. I didn't want to create a new mode because I don't want static users to be available for auto host user creation. The mode here won't be visible to the end user anyway, they'll just see the `teleport-static` group.

Reply:
I'd also lean towards a specific mode for `static` just to centralize the source of truth for how a user is being created. Adding a boolean to `HostUsersInfo` seems a bit less discoverable in the future than adding a possible value for `CreateHostUserMode`.

Would it be enough for role validation to fail, probably [here](https://github.com/gravitational/teleport/issues/46252), rather than building out a separate enum?